packages/react-components/src/components/chart/legend/useChartsLegend.tsx

```diff
@@ -22,6 +22,13 @@ import { LEGEND_NAME_MIN_WIDTH_FACTOR } from '../eChartsConstants';
 import Hide from './hide.svg';
```

Comment:
move the tanstack logic to a completely new file, so its clean and devoid of the cloudscape logic

Reply:
done